net/net_test.go

```diff
@@ -582,6 +582,34 @@ func TestGetIndexedIP(t *testing.T) {
 			expectError: false,
 			expectedIP:  "192.168.1.255",
 		},
+		{
+			cidr:        "255.255.255.0/24",
+			index:       256,
+			expectError: true,
+		},
+		{
+			cidr:        "fd:11:b2:be::/120",
+			index:       20,
+			expectError: false,
+			expectedIP:  "fd:11:b2:be::14",
+		},
+		{
+			cidr:        "fd:11:b2:be::/126",
+			index:       10,
+			expectError: true,
+		},
+		{
+			cidr:        "fd:11:b2:be::/120",
+			index:       255,
+			expectError: false,
+			expectedIP:  "fd:11:b2:be::ff",
+		},
+		{
+			cidr:        "00:00:00:be::/120",
+			index:       255,
+			expectError: false,
+			expectedIP:  "::be:0:0:0:ff",
```

Comment:
```
fmt.Println(net.ParseIP("00:00:00:be::ff"))
::be:0:0:0:ff
```
https://play.studygolang.com/p/YqkJBnMv2GP

Reply:
I guess this is a "tomato, tomato" thing, my understanding is that  you can choose where you want to use the `::`, it may be at the beginning or at the end of the address, the only restriction is that only can be used once 

>    2. Due to some methods of allocating certain styles of IPv6
>       addresses, it will be common for addresses to contain long strings
>       of zero bits.  In order to make writing addresses containing zero
>       bits easier, a special syntax is available to compress the zeros.
>       The use of "::" indicates one or more groups of 16 bits of zeros.
>       The "::" can only appear once in an address.  The "::" can also be
>       used to compress leading or trailing zeros in an address.

ref: https://tools.ietf.org/html/rfc4291#page-4

However, the recommendation is:

> 4.2.3.  Choice in Placement of "::"
>    When there is an alternative choice in the placement of a "::", the
>    longest run of consecutive 16-bit 0 fields MUST be shortened (i.e.,
>    the sequence with three consecutive zero fields is shortened in 2001:
>    0:0:1:0:0:0:1).  When the length of the consecutive 16-bit 0 fields
>    are equal (i.e., 2001:db8:0:0:1:0:0:1), the first sequence of zero
>    bits MUST be shortened.  For example, 2001:db8::1:0:0:1 is correct
>    representation.


ref: https://tools.ietf.org/html/rfc5952#page-10